### ISSUE TITLE
Issues training model with KD/logprobs

### ISSUE THREAD

**Issue by sankexin** [axolotl-ai-cloud/axolotl#2692]
### Please check that this issue hasn't been reported before.

- [x] I searched previous [Bug Reports](https://github.com/axolotl-ai-cloud/axolotl/labels/bug) didn't find any similar reports.

### Expected Behavior

train normally! train good result!


evolkit-logprobs-pipeline-75k-v2-sample is public by axolotl,but no detail code:

https://huggingface.co/datasets/axolotl-ai-co/evolkit-logprobs-pipeline-75k-v2


### Current behaviour
not good train result! anything wrong with my data of creating ?

### Steps to reproduce

distilabel-internal-testing/distilabel-example-instructions is created by distilabel library from tatsu-lab/alpaca! my code is :
```
from distilabel.pipeline import Pipeline
from distilabel.steps import LoadDataFromDicts, LoadDataFromHub
from distilabel.steps.tasks import TextGeneration
from distilabel.models import OpenAILLM

from transformers import AutoTokenizer

def extract_token_ids_and_logprobs(response, tokenizer_name="Qwen/Qwen3-8B"):
    tokenizer = AutoTokenizer.from_pretrained(tokenizer_name) 
    result = []

    if response == [[]]:
        return [result]


    for item in response[0]:
        encoded = tokenizer.encode(item["token"], add_special_tokens=False)
        token_id = encoded[0] if len(encoded) > 0 else None

        if token_id is None:
            continue

        result.append({
            "logprob": item["logprob"],
            "token": f"token_id:{token_id}"
        })

    return [result]


VLLM_BASE_URL = "http://0.0.0.0:8000/v1" 
MODEL_NAME = "Qwen/Qwen3-8B" 

with Pipeline() as pipeline:
    # Step 1: 
    '''
    load_data = LoadDataFromDicts(
        data=[{"prompt": "Write a poem about the sun and moon."}]
    )
    '''
    load_data = LoadDataFromHub(
        output_mappings={"instruction": "instruction"}  
    )
    '''
    load_data = LoadDataFromHub(
        repo_id="tatsu-lab/alpaca",
        split="train",
        batch_size=8,
        output_mappings={"instruction": "instruction"}
    )
    load_data.load()
    result = next(load_data.process())
    print(result)
    '''

    # Step 2:
    llm = OpenAILLM(
        model=MODEL_NAME,
        base_url=VLLM_BASE_URL,
        api_key="EMPTY",  
        max_retries=5,
        generation_kwargs={
            "temperature": 2.0,
            "max_new_tokens": 4096,
            "top_p": 0.9,
            "logprobs": True,       
            "top_logprobs": 20,     
            # "skip_special_tokens": True,
        }
    )

    # Step 3: 
    text_generation = TextGeneration(
        name="text_generation",
        llm=llm,
        input_batch_size=4,  
        # input_mappings={"instruction": "instruction"},
    )

    # Step 
    load_data >> text_generation

    '''
    # 
    llm.load()
    output = llm.generate_outputs(inputs=[[{"role": "user", "content": "Give three tips for staying healthy"}]])
    print(output)
    '''

if __name__ == "__main__":
    # pipeline
    distiset = pipeline.run(
        parameters={
            load_data.name: {
                "repo_id": "tatsu-lab/alpaca",  
                "split": "train", 
                "batch_size": 8,
            },
        }
    )

    # 
    distiset.save_to_disk("distilabel-internal-testing/distilabel-example-instructions")
    # distiset.push_to_hub(repo_id="xxx/distilabel-example-instructions-Qwen3-8B")
    # print(distiset)
    import pandas as pd
    for step_name in list(distiset.keys()):
        formatted_data = []
        step_output = distiset[step_name]["train"]
        formatted_data = []
        for item in step_output:
            item["llm_text_generation_logprobs"] = extract_token_ids_and_logprobs(item.get("llm_text_generation_logprobs", ""))
            messages_combined = [
                {"role": "user", "content": item.get("instruction", "")},
                {"role": "assistant", "content": item.get("generation", "")}
                ]
            item["messages"] = {"role": "user", "content": item.get("instruction", "")}
            item["messages_combined"] =  messages_combined
            formatted_data.append(item)

        df = pd.DataFrame(formatted_data)
        # print(df)
        df.to_parquet(f"tatsu-lab/{step_name}.parquet", engine="pyarrow")
        print(f"✅ {step_name}.parquet")


    '''
    from datasets import load_dataset
    ds = load_dataset("distilabel-internal-testing/distilabel-example-instructions")
    print(ds['train'][0])
    '''
```

### Config yaml

```yaml
# examples/qwen3/qlora-fsdp-kd_qwen3-8b.yaml:


base_model: Qwen/Qwen3-8B
# Automatically upload checkpoint and final model to HF
# hub_model_id: username/custom_model_name

plugins:
  - axolotl.integrations.kd.KDPlugin
kd_trainer: True
kd_ce_alpha: 0.1
kd_alpha: 0.9
kd_temperature: 1.0

torch_compile: False  # torch>=2.5.1, recommended to reduce vram

load_in_8bit: false
load_in_4bit: true
strict: false

datasets:
  - path: distilabel-internal-testing/distilabel-example-instructions
    type: axolotl.integrations.kd.chat_template
    field_messages: "messages_combined"
    logprobs_field: "llm_text_generation_vllm_logprobs"  # for kd only, field of logprobs
dataset_prepared_path:
val_set_size: 0.05
output_dir: ./outputs/out

sequence_len: 2048
sample_packing: true
eval_sample_packing: true
pad_to_sequence_len: true

adapter: qlora
lora_model_dir:
lora_r: 32
lora_alpha: 64
lora_dropout: 0.05
lora_target_linear: true

wandb_project:
wandb_entity:
wandb_watch:
wandb_name:
wandb_log_model:

gradient_accumulation_steps: 4
micro_batch_size: 1
num_epochs: 1
optimizer: adamw_torch_fused # adamw_torch
lr_scheduler: cosine
learning_rate: 0.0002

bf16: auto
tf32: false

gradient_checkpointing: true
gradient_checkpointing_kwargs:                                                                                                                                                                                                                      
  use_reentrant: false                                                                                                                                                                                                                              
resume_from_checkpoint:                                                                                                                                                                                                                             
logging_steps: 1                                                                                                                                                                                                                                    
flash_attention: true                                                                                                                                                                                                                               
                                                                                                                                                                                                                                                    
warmup_steps: 10                                                                                                                                                                                                                                    
evals_per_epoch: 4                                                                                                                                                                                                                                  
saves_per_epoch: 1                                                                                                                                                                                                                                  
weight_decay: 0.0                                                                                                                                                                                                                                   
fsdp:                                                                                                                                                                                                                                               
  - full_shard                                                                                                                                                                                                                                      
  - auto_wrap                                                                                                                                                                                                                                       
fsdp_config:                                                                                                                                                                                                                                        
  fsdp_limit_all_gathers: true                                                                                                                                                                                                                      
  fsdp_sync_module_states: true                                                                                                                                                                                                                     
  fsdp_offload_params: true                                                                                                                                                                                                                         
  fsdp_use_orig_params: false                                                                                                                                                                                                                       
  fsdp_cpu_ram_efficient_loading: true                                                                                                                                                                                                              
  fsdp_auto_wrap_policy: TRANSFORMER_BASED_WRAP                                                                                                                                                                                                     
  fsdp_transformer_layer_cls_to_wrap: Qwen3DecoderLayer                                                                                                                                                                                             
  fsdp_state_dict_type: FULL_STATE_DICT                                                                                                                                                                                                             
  fsdp_sharding_strategy: FULL_SHARD                                                                                                                                                                                                                
special_tokens:
```

### Possible solution

learn from  https://huggingface.co/datasets/axolotl-ai-co/evolkit-logprobs-pipeline-75k-v2

### Which Operating Systems are you using?

- [x] Linux
- [ ] macOS
- [ ] Windows

### Python Version

3.10

### axolotl branch-commit

main/a27b909  

### Acknowledgements

- [x] My issue title is concise, descriptive, and in title casing.
- [x] I have searched the existing issues to make sure this bug has not been reported yet.
- [x] I am using the latest version of axolotl.
- [x] I have provided enough information for the maintainers to reproduce and diagnose the issue.

**Comment by mhenrichsen** [comment]
@sankexin can you share your results including how you are inferencing?

**Comment by sankexin** [comment]
> [@sankexin](https://github.com/sankexin) can you share your results including how you are inferencing?

the loss get some negative, after some code, then get no negative，now loss normally!
[axolotl/src/axolotl/integrations/kd/trainer.py  at Line 76：
```
        if num_items_in_batch is None:
            num_items_in_batch = -1
```
There is no official code for creating kd logits  data, making it difficult to judge!
so, is my code of "geting logits dataset" correct?
